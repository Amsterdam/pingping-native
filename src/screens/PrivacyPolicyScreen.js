import React, { useState } from 'react';

import { useMutation } from '@apollo/client';
import PropTypes from 'prop-types';
import { StyleSheet, ScrollView, View } from 'react-native';

import testIDs from '../../e2e/modulesTestIDs';
import REGISTER_DEVICE_MUTATION from '../apollo/Mutation/registerDeviceMutation';
import routes from '../App/stacks/routes';
import FloppyDisk from '../assets/svg/FloppyDisk';
import FilledHeader from '../components/header/FilledHeader';
import Header from '../components/header/Header';
import Container from '../components/shared/Container';
import Loading from '../components/shared/LoadingComponent';
import PrivacyPolicyAccordion from '../components/shared/PrivacyPolicyAccordion';
import Button from '../components/shared/RoundedButton';
import Body from '../components/typography/Body';
import Title from '../components/typography/Title';
import { ONBOARDING_STATES } from '../config/constants';
import theme from '../config/theme';
import { getFromAsyncStorage, multiSetAsyncStorage } from '../helpers/asyncStorageHelpers';
import { doRegisterDevice } from '../helpers/authHelper';
import sentryHelper from '../helpers/sentryHelper';

// this screen is used in both the onboardingscreen and the settingscreen
// therefore we configure it based on if the user has accepted the privacy policy
// we need to set the color accordingly

function PrivacyPolicyScreen({ navigation, route }) {
	const { fromOnboarding } = route.params || { fromOnboarding: false };
	const [open, setOpen] = useState(false);
	const [registerDevice] = useMutation(REGISTER_DEVICE_MUTATION);
	const [loading, setLoading] = useState(false);

	const toggleOpen = () => {
		setOpen(!open);
	};

	const doAcceptPolicy = async () => {
		setLoading(true);
		try {
			await doRegisterDevice(registerDevice);
			await multiSetAsyncStorage([
				['@pingpingNative_acceptedPolicy', JSON.stringify(true)],
				['@pingpingNative_onboardingStatus', ONBOARDING_STATES.onboardingQuestionsStarted],
			]);
			const token = await getFromAsyncStorage('@pingpingNative_accessToken');
			if (token) {
				setLoading(false);
				navigation.navigate(routes.onboardingStack.screens.questionScreen);
			}
		} catch (error) {
			sentryHelper(error.message);
		}
	};

	return (
		<Container
			testID={testIDs.PRIVACY.SCREEN}
			statusBarColor={fromOnboarding ? theme.colors.white : theme.colors.headerColor}
		>
			{fromOnboarding ? (
				<Header title="Privacy" color="light" />
			) : (
				<FilledHeader navigation={navigation} title="Privacy" />
			)}
			<ScrollView contentContainerStyle={styles.content}>
				<View style={styles.viewContainer}>
					<View>
						<FloppyDisk />
					</View>
					<View>
						<Title style={styles.title} variant="h1" align="center">
							PRIVACY
						</Title>
						<Body variant="b3" style={styles.onboardingText} align="center">
							Om Ping Ping optimaal te laten functioneren verzamelen wij informatie.
							Klik hieronder om meer hierover te lezen. Wij slaan zo min mogelijk
							informatie op.
						</Body>
					</View>
					<PrivacyPolicyAccordion open={open} toggleOpen={toggleOpen} />
					{fromOnboarding && (
						<View style={styles.buttonContainer}>
							<Button
								testid={testIDs.PRIVACY.PRIVACY_ACCEPT_BUTTON}
								onPress={doAcceptPolicy}
								label="Accepteren"
							/>
						</View>
					)}
				</View>
			</ScrollView>
			{loading && <Loading />}
		</Container>
	);
}

const styles = StyleSheet.create({
	viewContainer: {
		flex: 1,
		justifyContent: 'space-evenly',
		alignItems: 'center',
		backgroundColor: theme.colors.background,
		padding: theme.spacing.s,
	},
	title: {
		color: theme.colors.text,
		marginBottom: theme.spacing.m,
	},
	onboardingText: {
		color: theme.colors.subText,
	},
	buttonContainer: {
		alignItems: 'center',
	},
	content: {
		flexGrow: 1,
	},
});

PrivacyPolicyScreen.propTypes = {
	navigation: PropTypes.object.isRequired,
	route: PropTypes.object.isRequired,
};

export default PrivacyPolicyScreen;
