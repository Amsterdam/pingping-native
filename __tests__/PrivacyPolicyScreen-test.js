import React from 'react';

import { MockedProvider } from '@apollo/client/testing';
import { NativeBaseProvider } from 'native-base';
import renderer from 'react-test-renderer';

import PrivacyPolicyScreen from '../src/screens/PrivacyPolicyScreen';

jest.useFakeTimers();
const navigation = { navigate: jest.fn() };

test('renders correctly', () => {
	const tree = renderer
		.create(
			<NativeBaseProvider>
				<MockedProvider mocks={[]}>
					<PrivacyPolicyScreen
						navigation={navigation}
					/>
				</MockedProvider>
			</NativeBaseProvider>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
