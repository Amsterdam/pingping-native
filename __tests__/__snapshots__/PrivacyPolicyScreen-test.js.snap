// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
  testID="PRIVACY_SCREEN"
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#0d2036",
        }
      }
    >
      <View
        noShadow={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "paddingTop": undefined,
              "right": 0,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
              "top": 0,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#0d2036",
            },
          ]
        }
        transparent={true}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 4,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID="HEADER_BACK_BUTTON"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 34,
                      "marginLeft": 1,
                      "marginRight": 2,
                      "marginTop": 0,
                      "paddingTop": 1,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ],
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="AntDesign"
            >
              î˜œ
            </Text>
          </View>
        </View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "Heavitas",
                "fontSize": 30,
                "textAlign": "left",
              },
              Object {
                "color": "#fff",
                "fontSize": 14,
                "marginTop": 5,
              },
            ]
          }
        >
          Privacy
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          Object {
            "flexGrow": 1,
          },
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "space-evenly",
              "padding": 15,
            }
          }
        >
          <View>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={160}
              bbWidth={160}
              fill="none"
              focusable={false}
              height={160}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 160,
                    "width": 160,
                  },
                ]
              }
              vbHeight={160}
              vbWidth={160}
              width={160}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M145.238 2H21.606c-1.524 0-3.047.61-4.114 1.727L3.727 17.492A5.81 5.81 0 002 21.606v123.632c0 5.029 4.114 9.143 9.143 9.143h134.095c5.029 0 9.143-4.114 9.143-9.143V11.143c0-5.029-4.114-9.143-9.143-9.143z"
                  fill={4285821304}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGCircle
                  cx={78.191}
                  cy={80.73}
                  fill={4283518549}
                  opacity={0.43}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  r={64.356}
                  stroke={4282794825}
                  strokeWidth={2.032}
                />
                <RNSVGPath
                  clipRule={0}
                  d="M145.238 2H21.606c-1.524 0-3.047.61-4.114 1.727L3.727 17.492A5.81 5.81 0 002 21.606v123.632c0 5.029 4.114 9.143 9.143 9.143h134.095c5.029 0 9.143-4.114 9.143-9.143V11.143c0-5.029-4.114-9.143-9.143-9.143z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  clipRule={0}
                  d="M145.238 2H21.606c-1.524 0-3.047.61-4.114 1.727L3.727 17.492A5.81 5.81 0 002 21.606v123.632c0 5.029 4.114 9.143 9.143 9.143h134.095c5.029 0 9.143-4.114 9.143-9.143V11.143c0-5.029-4.114-9.143-9.143-9.143z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGCircle
                  cx={78.191}
                  cy={80.73}
                  fill={4291414473}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  r={21.333}
                />
                <RNSVGCircle
                  cx={78.191}
                  cy={80.73}
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  r={21.333}
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M131.016 154.381H25.365v-12.698c0-5.029 4.114-9.143 9.143-9.143h87.365c5.028 0 9.143 4.114 9.143 9.143v12.698z"
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M131.016 154.381H25.365v-12.698c0-5.029 4.114-9.143 9.143-9.143h87.365c5.028 0 9.143 4.114 9.143 9.143v12.698z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M137.873 132.54h7.111v7.111h-7.111z"
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M137.873 132.54h7.111v7.111h-7.111z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M99.524 125.784h34.032"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={4.063}
                />
                <RNSVGPath
                  d="M75.397 77.683h4.571v4.571h-4.571z"
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M75.397 77.683h4.571v4.571h-4.571z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M78.608 88.907l9.587-3.362 1.68 4.794-9.586 3.362z"
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M78.608 88.907l9.587-3.362 1.68 4.794-9.586 3.362z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M154.381 33.492c-3.352 0-6.095-2.743-6.095-6.095 0-3.353 2.743-6.095 6.095-6.095v12.19z"
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M154.381 33.492c-3.352 0-6.095-2.743-6.095-6.095 0-3.353 2.743-6.095 6.095-6.095v12.19z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M2 33.492c3.352 0 6.095-2.743 6.095-6.095 0-3.353-2.743-6.095-6.095-6.095v12.19z"
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M2 33.492c3.352 0 6.095-2.743 6.095-6.095 0-3.353-2.743-6.095-6.095-6.095v12.19z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGCircle
                  cx={13.022}
                  cy={41.365}
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  r={4.317}
                />
                <RNSVGCircle
                  cx={140.26}
                  cy={41.365}
                  fill={4282465873}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  r={4.317}
                />
                <RNSVGPath
                  d="M124.413 48.73H54.317c-5.028 0-9.142-4.114-9.142-9.143V2h88.381v37.587c0 5.08-4.115 9.143-9.143 9.143z"
                  fill={4283518549}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M102.063 48.73H57.365c-5.029 0-9.143-4.114-9.143-9.143V2h62.984v37.587c0 5.08-4.114 9.143-9.143 9.143z"
                  opacity={0.48}
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  clipRule={0}
                  d="M124.413 48.73H54.317c-5.028 0-9.142-4.114-9.142-9.143V2h88.381v37.587c0 5.08-4.115 9.143-9.143 9.143z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M108.159 8.603h17.016"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={4.063}
                />
                <RNSVGPath
                  d="M99.016 48.73H54.317c-5.028 0-9.142-4.114-9.142-9.143V2h62.984v37.587c0 5.08-4.115 9.143-9.143 9.143z"
                  fill={4290757056}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M82 48.73l26.159-36.114v-8.28L75.194 48.73H82zM55.994 48.578c.61.101-2.337.203-1.676.203h15.136L104.35 2.05H90.737L55.994 48.578z"
                  fill={4292730333}
                  opacity={0.52}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M99.016 48.73H54.317c-5.028 0-9.142-4.114-9.142-9.143V2h62.984v37.587c0 5.08-4.115 9.143-9.143 9.143z"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
                <RNSVGPath
                  d="M56.654 7.587h12.191v29.46H56.654z"
                  fill={4283518549}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M56.654 7.587h6.806v29.46h-6.806z"
                  fill={4282465873}
                  opacity={0.48}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M56.654 7.587h12.191v29.46H56.654zM95.511 7.587h6.095v3.048h-6.095zM13.835 154.381V142.19"
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4282465873}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={3.048}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View>
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Heavitas",
                    "fontSize": 30,
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 28,
                    "fontWeight": "400",
                    "marginBottom": 20,
                    "textAlign": "center",
                  },
                ]
              }
            >
              PRIVACY
            </Text>
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Object {
                  "color": undefined,
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 14,
                  "lineHeight": 25,
                  "textAlign": "center",
                }
              }
            >
              Om Ping Ping optimaal te laten functioneren verzamelen wij informatie. Klik hieronder om meer hierover te lezen. Wij slaan zo min mogelijk informatie op.
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderColor": "#000",
                "borderWidth": 1,
                "margin": 15,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 10,
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "fontFamily": "Heavitas",
                      "fontSize": 30,
                      "textAlign": "left",
                    },
                    Object {
                      "fontSize": 12,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Privacy Policy
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "fontSize": 16,
                        "marginLeft": 8,
                      },
                    ],
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                type="AntDesign"
              >
                î˜†
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
